mypy does not error when calling a `TypeAliasType`, if it refers to `dict`

**Bug Report**

If I declare a type with `type T = dict`, then instantiate it with `T()`, mypy does not error.

I spot-checked some other types like `type T = set`, `type T = int` etc, and mypy complains about these when called. Only `type T = dict` seems to wrongly pass.

**To Reproduce**

```python
type T = dict
x: T = T()
```

**Expected Behavior**

```shell
$ mypy t.py
t.py:2: error: "TypeAliasType" not callable  [operator]
Found 1 error in 1 file (checked 1 source file)
```

**Actual Behavior**

```shell
$ mypy t.py
Success: no issues found in 1 source file
```

**Your Environment**

- Mypy version used: `mypy 1.17.1 (compiled: yes)`
- Mypy command-line flags: `mypy`
- Mypy configuration options from `mypy.ini` (and other config files): None
- Python version used: `Python 3.13.5`

--- a/mypy/semanal.py
+++ b/mypy/semanal.py
@@ -7678,7 +7678,7 @@ def refers_to_fullname(node: Expression, fullnames: str | tuple[str, ...]) -> bo
         return False
     if node.fullname in fullnames:
         return True
-    if isinstance(node.node, TypeAlias):
+    if isinstance(node.node, TypeAlias) and not node.node.python_3_12_type_alias:
         return is_named_instance(node.node.target, fullnames)
     return False
 

--- a/test-data/unit/check-python312.test
+++ b/test-data/unit/check-python312.test
@@ -2095,3 +2095,19 @@ from dataclasses import dataclass
 class Test[*Ts, R]:
     a: Callable[[*Ts], R]
 [builtins fixtures/dict.pyi]
+
+[case testPEP695AliasDoesNotReferToFullname]
+# https://github.com/python/mypy/issues/19698
+from typing import TypeAliasType
+type D = dict
+type T = type
+type TA = TypeAliasType
+
+D()          # E: "TypeAliasType" not callable
+X = TA("Y")  # E: "TypeAliasType" not callable
+
+x: object
+if T(x) is str:     # E: "TypeAliasType" not callable
+    reveal_type(x)  # N: Revealed type is "builtins.object"
+[builtins fixtures/tuple.pyi]
+[typing fixtures/typing-full.pyi]

